### DISCUSSION TITLE
Experiences of Surface Go 3 on linux?

### DISCUSSION THREAD

**Discussion by cherryband** [linux-surface/linux-surface#1192] [Q&A] [answered]
Hey, I'm considering buying SGO3 specifically to run linux. But, there are nearly zero information about this particular model on the internet. I intend to use it for note taking, watching online videos and drawing if it works well.

* The feature matrix indicates that hibernation for this model is unsupported. What are the implications of this? Is it like most linux laptops where you can only do powered standby?
* What kind of distros/DEs have you used, and what fares better for tablet use?
* Does USB-C charging work reliably? If it doesn't, is there a workaround?
* Is the pen fully functional, e.g., with pressure sensitivity, palm rejection, eraser and tilt support? How is the input lag?
* Are there any other idiosyncracies aside from those mentioned in the matrix?

I appreciate the hard work went into this to make linux on surface tablets a possibility. Thanks!

**Comment by cherryband** [comment, marked as answer]
Now that I've used this device for a month, I think I could answer this myself. I'm leaving it here for others.

* I was right with the hibernation; practically it doesn't have much impact, unless you leave it for weeks on idle. Suspend works well.
* I've tried mainly GNOME and Phosh, and even though Phosh is more performant, GNOME just has more features and has the best gesture support while also working well with a keyboard. For my purpose Fedora offered the best balance of out-of-the-box experience and up-to-date software. Others I've tried, notably Debian, Alpine Linux, and OpenSUSE, all fell behind either or both of these aspects.
* See #1222 
* I was surprised to find that all functions of the Surface pen is supported, even without custom kernel/driver. In fact with an up-to-date kernel there was no need to install linux-surface.
* There are touch issues after suspend, but it's not a deal-breaker. The touch sometimes doesn't work until you use the Surface Pen(#1223).

**Reply by mavaki** [reply]
I'm thinking of getting the Go 3 to run Linux, and I have a few questions about your experience if you're able to share. I can't seem to find much information on the Go 3, so your original post was already very helpful!

- Are you still using Fedora for your Go 3?
- Did you have to do anything besides the basic install to get everything working properly (besides usb-c and suspend touch issues)?
- Is there anything abnormal during the booting process, or is it just like any other computer?
- Have you tried drawing/taking notes on it (with xournal++ or any other app)?

I would love to use it to take notes and watch videos, so any general advice on that would be great too. Thanks so much!

**Comment by cherryband** [comment]
* I just decided to try out FydeOS, but Fedora has been installed on it for the longest.
* I didn't even need a linux-surface kernel; everything that should work worked out of the box.
* The firmware shows an on screen keyboard button when booting, but only before loading the OS. If you encrypt your Fedora install, you'll need a keyboard to enter the password.
* I've both drawn with Krita and took notes on Xournal++. Touch rejection is a problem: I frequently had problems in Krita where the touch isn't recognised even after removing the pen. And it's sometimes slow to recognise how many fingers I've touched. But I personally had a good time: eraser tip works on both programs, and as a display tablet it was decent enough to get a work done. I personally didn't take notes that much but it's doable.

19. März 2024, 23:06 von ***@***.***:

**Comment by cherryband** [comment]
Update after installing several more OSes:
* With recent kernel, you do not need linux-surface. Tails works on the machine, including touch and pen support.
* On the latest firmware, Fedora ISO does not boot. I've tried versions down to 37, but all of them hang after GRUB.
* FydeOS for Surface GO 3 is OK if you want a touch friendly experience, but you will miss some features.
  * Keyboard cover deactivation: the surface knows if the type cover is folded back and treat it like it's disconnected. FydeOS doesn't do that, meaning you will have to disconnect it or manually toggle the on screen keyboard. This also affects switching to tablet mode.
  * Pen eraser: pressure sensitivity works, but the eraser tip will behave as if it was the main tip.
  * Performance: the Surface runs very hot on FydeOS and the battery runs out in 2 hours.
* Now I recommend Alpine Linux, because the feature is well supported and it's very lightweight. I get excellent battery life and very little lags. Some tips for setting up:
  * Plasma 6 is (imo) still not ready for tablet use; stick to GNOME unless you're fine with not having gesture controls.
  * Because the installation is bare minimum, you will need to install and configure following modules to get the full support:
    * Install and configure [GNOME](https://wiki.alpinelinux.org/wiki/GNOME), [PipeWire](https://wiki.alpinelinux.org/wiki/PipeWire), [NetworkManager (conflicts with default networking package)](https://wiki.alpinelinux.org/wiki/NetworkManager), [Bluetooth](https://wiki.alpinelinux.org/wiki/Bluetooth), [PipeWire Bluetooth service](https://wiki.alpinelinux.org/wiki/PipeWire#Bluetooth_audio), [Graphics driver](https://wiki.alpinelinux.org/wiki/Intel_Video), and `linux-firmware`.
    * For ambient light sensor and accelerometer, install `iio-sensor-proxy` package and enable the service with `doas rc-update add iio-sensor-proxy`.
    * For languages other than English, see https://wiki.alpinelinux.org/wiki/Post_installation#Language_support.
    * Follow [this guide](https://wiki.alpinelinux.org/wiki/UEFI_Secure_Boot) to setup secure boot. You need to enrol the key from the OS; there's no such option in the firmware interface.
    * By default, Alpine Linux will turn off when you press the power button, as it does not see the "lid" at `/proc/acpi/button/lid/LID`(it's at `/proc/acpi/button/lid/LID0` instead), behaving like a desktop. GNOME setting doesn't override this. Follow [this guide](https://wiki.alpinelinux.org/wiki/Configure_action_when_power-button_is_pressed) to make it suspend. Also consider changing the lid path to `/proc/acpi/button/lid/LID0` (and `button/lid:*:close:*` to `button/lid0:*:close:*`) in `/etc/acpi/handler.sh`. This will suspend (and lock!) the device when you close the type cover.

**Comment by mavaki** [comment]
Thanks for the update by the way! I might eventually try Alpine Linux, but I've been pretty happy with Arch Linux on my Surface Go 3 (without the linux-surface kernel). I have recently had a problem with Xournal++ where it takes me to the login screen when I try to save a file. I have started to save files as soon as I open them so that I can avoid losing notes. The pen works with eraser as well, although there is a small box that can sometimes stay on the screen after erasing until you start writing again. It hasn't really bothered me though. I would say that the one thing I want to fix soon involves the on-screen keyboard: when I click the backspace in some applications (such as Firefox), it defaults to selecting the letter before the cursor instead of deleting it, which makes deleting/retyping things very annoying.

**Comment by cherryband** [comment]
Another one: if you don't mind going through the hoops with manual flashing, postmarketOS offers all the benefits of Alpine Linux without a need to hunt for packages or struggle with configurations. Also consider Gnome Mobile; it's not stable, but it's the closest thing you can get to a full convertible experience on Linux.
